[ADD] sign_stamp: add company stamp support in Sign module

utpat-odoo opened the pull request:
[ADD] sign_stamp: add company stamp support in Sign module

Purpose:
Add a new Stamp sign item type to the Sign module, allowing companies to apply an official stamp containing company details and logo when validating documents.

Technical Changes:
data:
Added new sign.item.type entry for stamp in sign_data.xml with placeholder.
  
views:
Extended sign._doc_sign to inject company-related fields (company, address, city, country, VAT) into the DOM.
Introduced sign_request_templates.xml for provide company details to the frontend.
  
Frontend:
Patched document_signable.js to fetch company details from the DOM and pass them to the iframe.
Extended signable_PDF_iframe.js to: Detect stamp items, open a custom stamp dialog, handle confirm/cancel actions
  
Dialogs & Signature Rendering:
Added StampSignDetailsDialog for capturing stamp information and logo upload.
Extended NameAndSignature to generate a dynamic SVG stamp using company data and optional logo.
Ensured real-time preview updates on input changes.

utpat-odoo committed:
[ADD] sign_stamp: add stamp field in sign app

Purpose:
The Stamp field allows users to apply an official company stamp during the
document signing process.

Technical changes:
add new stamp field in sign.item.type
Configured a multi-line placeholder to display structured stamp information
(Company, Address, City, Country, VAT Number)

robodoo commented:
[![Pull request status dashboard](https://mergebot.odoo.com/odoo/tutorials/pull/1081.png)](https://mergebot.odoo.com/odoo/tutorials/pull/1081)

utpat-odoo committed:
[IMP] sign_stamp: add view for input fields

Technical Changes:
inherit sign._doc_sign template and add input fields for company, address,
city, country, VAT Number

utpat-odoo committed:
[IMP] sign_stamp: add company stamp support for users

Technical Changes:
- stamp_sign_stamp and stamp_sign_stamp_frame field in res.users
- Implemented  and  methods in
   to fetch user stamp and frame.
- Allows displaying the company stamp and frame in signature dialogs for sign
  requests.

utpat-odoo committed:
[IMP] sign_stamp: add test dialog box and fetch company data for auto-fill

Technical Changes:
add stamp_sign_stamp and stamp_sign_stamp_frame in res.users for
store stamp data
fetch company data in sign_request page
add test dialog box
when user click stamp then test dialog box is open

utpat-odoo committed:
[IMP] sign_stamp: add company details fields in dialog box fields are auto-fill

- add Input fields in dialog box
- Extend getIframeProps to include companyInfo in iframe props
- Pass company details (name, address, city, country, VAT)
  when opening stamp dialog
- Prefill stamp adoption dialog inputs with company information

utpat-odoo committed:
[IMP] sign_stamp: auto-fill stamp and update according to dialog box details

user can see company information in stamp default
when user click stamp then dialog box open and user can edit company information

utpat-odoo committed:
[IMP] sign_stamp: add preview box for stamp

add preview box for stamp

utpat-odoo committed:
[IMP] sign_stamp: add sign and sign all button in stamp dialog

add sign and sign all button in stamp dialog

bit-odoo commented on sign_stamp/controllers/main.py:
```diff
@@ -0,0 +1,42 @@
+from odoo import http
+from odoo.addons.sign.controllers.main import Sign
+import logging
```
External import should be first, and then after odoo improt in alphabetical order.

bit-odoo commented on sign_stamp/controllers/main.py:
```diff
@@ -0,0 +1,42 @@
+_logger = logging.getLogger(__name__)
+
+
+class Sign(Sign):
```
Should be one empty line after it.

bit-odoo reviewed (commented):
Hello,
I think the stamp should be single, so sing all is not required for this.
Thanks

utpat-odoo committed:
[IMP] sign_stamp: remove signall button from stamp

remove signall button from stamp and also remove onConfirmAll function
Now primary button is Sign

utpat-odoo committed:
[IMP] sign_stamp: add company stamp support in Sign module

Purpose:
Add a new Stamp sign item type to the Sign module, allowing companies to apply
an official stamp containing company details and logo when validating documents.

Technical Changes:
1.data:
-Added new sign.item.type entry for stamp in sign_data.xml with placeholder.

2.views:
- Extended sign._doc_sign to inject company-related fields (company, address,
city, country, VAT) into the DOM.
- Introduced sign_request_templates.xml for provide company details to the
frontend.

3.Frontend:
- Patched document_signable.js to fetch company details from the DOM and pass
them to the iframe.
- Extended signable_PDF_iframe.js to: Detect stamp items, open a custom stamp
dialog, handle confirm/cancel actions

4.Dialogs & Signature Rendering:
- Added StampSignDetailsDialog for capturing stamp information and logo upload.
- Extended NameAndSignature to generate a dynamic SVG stamp using company data
and optional logo.
- Ensured real-time preview updates on input changes.

utpat-odoo committed:
[IMP] sign_stamp: add company name in stamp field

Purpose:
Ensure that stamp sign items display the current company name
automatically, providing consistent and company-aware labeling
when using stamps in documents.

Technical Changes:
- Added a sequence field to control priority and ensure stamp items
  with the highest priority appear at the top.
- Extended the sign.item.type model to compute the stamp name
  using the active company context.
- Implemented the _compute_display_name method to dynamically
  assign the company name to stamp items.
